src/app/ReadHandler.h

```diff
@@ -166,20 +166,19 @@ class ReadHandler : public Messaging::ExchangeDelegate
     public:
         virtual ~Observer() = default;
 
-        /// @brief Callback invoked to notify a ReadHandler was created and can be registered
-        /// @param[in] apReadHandler  ReadHandler getting added
-        virtual void OnReadHandlerCreated(ReadHandler * apReadHandler) = 0;
-
-        /// @brief Callback invoked when a ReadHandler went from a non reportable state to a reportable state so a report can be
-        /// sent immediately if the minimal interval allows it. Otherwise the report should be rescheduled to the earliest time
-        /// allowed.
-        /// @param[in] apReadHandler  ReadHandler that became dirty
+        /// @brief Callback invoked to notify a subscription was successfully established for a read handler
+        /// @param[in] apReadHandler  ReadHandler that completed its subscription
+        virtual void OnReadHandlerSubscribed(ReadHandler * apReadHandler) = 0;
+
+        /// @brief Callback invoked when a ReadHandler went from a non reportable state to a reportable state. Indicates to the
+        /// observer that a report should be emitted when the min interval allows it.
```

Comment:
```suggestion
        /// observer that a report should be emitted when the min interval allows it.
        ///
        /// This will only be invoked for subscribe-type ReadHandler objects, and only after
        /// OnSubscriptionEstablished has been called.
```